### PR TITLE
[oceanbase] add jdbc options and support oracle mode

### LINKED ISSUE #983: [oceanbase] Allow user to pass custom JDBC URL parameters
As we use jdbc connection for snapshot reading, we should allow user to pass custom jdbc url parameters on startup.

Ref: #921

### LINKED ISSUE #1309: [oceanbase] compatible with mysql 8.0 driver
The OceanBase CDC connector uses mysql driver for reading snapshot data, and now only `mysql-connector-java 5.1.47` has been fully tested. More later drivers  like `8.0.27` should be supported in the future version.

This work might be related to #980 and #983.

### LINKED ISSUE #1777: [oceanbase] Support OceanBase Oracle Mode
**Is your feature request related to a problem? Please describe.**
Support OceanBase (Enterprise Edition) Oracle Mode in OceanBase CDC connector.

**Describe the solution you'd like**
Use https://github.com/oceanbase/obconnector-j to deal with the snapshot reading.

**Describe alternatives you've considered**
No.

### PR DESCRIPTION
**Pull request by whhe**
Use oceanbase official jdbc driver to support oracle mode and add jdbc properties, close #983, close #1777.

### CODE REVIEW

**Comment by whhe** [comment]
Seems that the driver is under LGPL, so I need to adjust the introduction of it. WIP.

**Comment by whhe** [comment]
Now the connector parses ByteString to String with real encoding of field befor the process in runtime converter, fix #2114.

File: flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/source/OceanBaseConnection.java [hunk 1]
```diff
@@ -16,29 +16,47 @@
...
-    protected static final String URL_PATTERN =
-            "jdbc:mysql://${hostname}:${port}/?useInformationSchema=true&nullCatalogMeansCurrent=false&useSSL=false&useUnicode=true&characterEncoding=UTF-8&characterSetResults=UTF-8&zeroDateTimeBehavior=convertToNull&connectTimeout=${connectTimeout}";
-    protected static final String DRIVER_CLASS_NAME = "com.mysql.jdbc.Driver";
+    private static final Properties DEFAULT_JDBC_PROPERTIES = initializeDefaultJdbcProperties();
+    private static final String MYSQL_URL_PATTERN =
+            "jdbc:mysql://${hostname}:${port}/?connectTimeout=${connectTimeout}";
+    private static final String OB_URL_PATTERN =
+            "jdbc:oceanbase://${hostname}:${port}/?connectTimeout=${connectTimeout}";
+
+    private String compatibleMode;
```
Comment (ruanhang1993) [inline flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/source/OceanBaseConnection.java:43]: Could this `compatibleMode` be final and decided in the constructor?

File: flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/source/OceanBaseConnection.java [hunk 2]
```diff
@@ -49,7 +67,111 @@ public static Configuration config(
...
-    public static JdbcConnection.ConnectionFactory factory(ClassLoader classLoader) {
-        return JdbcConnection.patternBasedFactory(URL_PATTERN, DRIVER_CLASS_NAME, classLoader);
+    private static String formatJdbcUrl(String jdbcDriver, Properties jdbcProperties) {
+        Properties combinedProperties = new Properties();
+        combinedProperties.putAll(DEFAULT_JDBC_PROPERTIES);
+        if (jdbcProperties != null) {
+            combinedProperties.putAll(jdbcProperties);
+        }
+        String urlPattern =
+                jdbcDriver.toLowerCase().contains("oceanbase") ? OB_URL_PATTERN : MYSQL_URL_PATTERN;
```
Comment (ruanhang1993) [inline flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/source/OceanBaseConnection.java:77]: If `compatibleMode` could be decided in the constructor, this condition could be like `compatibleMode.equals(xxx)`.

**Review by ruanhang1993 — APPROVED** [review]
LGTM

**Pull request merged**